content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,266 @@
+
+2. Consider how we might define more patterns to keep the "Save" button easily accessible on long forms:
+- add save and cancel buttons on top AND bottom?
+- use a sticky header with save and cancel buttons?
+-->
+
+#### Adjacent to inline editing input
+
+To simplify the interface, some content may be edited inline without taking the user to a separate flow.
+
+An input used for inline editing should have a save button placed very close to visually connect the input with the button.
+
+<img
+    width="960"
+    alt="Two images of a pull request header. The first image shows the title as read-only text. The second image shows a text input to edit the pull request title with 'Cancel' and 'Save' buttons below the input."
+    src="https://user-images.githubusercontent.com/2313998/151623718-9f3f7fe5-ad35-4e4a-b236-da594388203b.png"
+/>
+
+#### Other
+
+If you believe your submit button will not be intuitive or easily accessible using any of the placement options listed here, you may consider alternatives and optionally propose a new Primer pattern.
+
+## Explicit saving
+
+When designing a form, start with an explicit saving pattern. Avoid mixing explicit and automatic save patterns on a single page with multiple forms, and never mix save patterns in a single form. For example: if I upload a profile photo in the "Public profile" form, that change should not be saved until I explicitly submit the whole form.
+
+Explicitly saved forms can be saved by clicking a submit button or by pressing Enter while focused on an input.
+
+If there is an error saving the data, the user's data should be preserved in the form and they should be given [feedback](#feedback) about the failure.
+
+<DoDontContainer>
+  <Do>
+    <img alt="A form broken into two sections where one section has a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887935-980adacd-41aa-47cb-8756-eeaa1a6d0f2b.png" />
+    <Caption>Separate auto-saving inputs from an explicitly saved form</Caption>
+  </Do>
+  <Dont>
+    <img alt="A form with a single section and it contains a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887936-d41900ca-2d63-43ac-bf82-55b4bba0df61.png" />
+    <Caption>Don't mix auto-saving inputs and explicitly saved inputs in the same form</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Unsaved changes
+
+If a user tries to navigate away from a page with a form that has unsaved changes, you have the option to warn them that their changes will be lost using the [`beforeunload` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). Modern browsers prevent developers from customizing the `alert` message, so we're limited to a generic message like "Do you want to leave this site? Changes you made may not be saved."
+
+### Declarative controls
```

Comment:
Should this say **Declarative inputs** instead?

Reply:
We switched to calling them "controls"